src/components/content-tab-import-export.tsx

```diff
@@ -86,20 +99,170 @@ export const ExportSite = ( {
 				</p>
 			</div>
 			<div className="gap-4 flex flex-row">
-				<Button onClick={ onExportFullSite } variant="primary">
+				<CoreButton onClick={ onExportFullSite } variant="primary">
 					{ __( 'Export entire site' ) }
-				</Button>
-				<Button onClick={ onExportDatabase } type="submit" variant="secondary">
+				</CoreButton>
+				<CoreButton onClick={ onExportDatabase } type="submit" variant="secondary">
 					{ __( 'Export database' ) }
-				</Button>
+				</CoreButton>
 			</div>
 		</div>
 	);
 };
 
+const InitialImportButton = ( {
+	children,
+	isInitial,
+	openFileSelector,
+}: {
+	children: React.ReactNode;
+	isInitial: boolean;
+	openFileSelector: () => void;
+} ) =>
+	isInitial ? (
+		<Button
+			variant="icon"
+			className="w-full [&>div.border-zinc-300]:hover:border-a8c-blueberry"
+			onClick={ openFileSelector }
+		>
+			{ children }
+		</Button>
+	) : (
+		<div className="w-full">{ children }</div>
+	);
+
+const ImportSite = ( props: { selectedSite: SiteDetails } ) => {
+	const { __ } = useI18n();
+	const { importFile, updateSite, startServer, loadingServer } = useSiteDetails();
+	const importConfirmation = useConfirmationDialog( {
+		message: sprintf( __( 'Overwrite %s?' ), props.selectedSite.name ),
+		checkboxLabel: __( "Don't ask again" ),
+		detail: __( 'Importing a backup will replace the existing files and database for your site.' ),
+		confirmButtonLabel: __( 'Import' ),
+		localStorageKey: 'dontShowImportConfirmation',
+	} );
+
+	const { dropRef, isDraggingOver } = useDragAndDropFile< HTMLDivElement >( {
+		onFileDrop: ( file: File ) => {
+			importConfirmation( () => importFile( file, props.selectedSite ) );
+		},
+	} );
+	const inputFileRef = useRef< HTMLInputElement >( null );
+	const openFileSelector = async () => {
+		inputFileRef.current?.click();
+	};
+	const onFileSelected = async ( e: React.ChangeEvent< HTMLInputElement > ) => {
+		const file = e?.target?.files?.[ 0 ];
+		if ( ! file ) {
+			return;
+		}
+		importConfirmation( async () => {
+			await importFile( file, props.selectedSite );
+			clearImportFileInput();
+		} );
+	};
+	const openSite = async () => {
+		if ( ! props.selectedSite.running ) {
+			speak( __( 'Starting the server before opening the site link' ) );
+			await startServer( props.selectedSite.id );
+		}
+		getIpcApi().openSiteURL( props.selectedSite.id );
+	};
+	const clearImportFileInput = () => {
+		if ( inputFileRef.current ) {
+			inputFileRef.current.value = '';
+		}
+	};
+	const clearImportState = () => {
+		updateSite( { ...props.selectedSite, importState: ImportState.Initial } );
+		clearImportFileInput();
+	};
+
+	const startLoadingCursorClassName =
+		loadingServer[ props.selectedSite.id ] && 'animate-pulse duration-100 cursor-wait';
+
+	const isImporting = props.selectedSite.importState === ImportState.Importing && ! isDraggingOver;
```

Comment:
By adding a fake delay in the import process, I noticed that we allow importing a file while another import is going on. Probably this is caused by this condition, as `isDraggingOver` will make `isImporting` false.

Unless we want to explicitly allow interrupting an import, which I presume it's not the case, we could try removing `&& ! isDraggingOver` from the condition.

https://github.com/user-attachments/assets/3a724dd3-39bf-48c5-836d-c6dde8e381ee

Reply:
Great catch! 
Changed https://github.com/Automattic/studio/pull/369/commits/9153649af06c7b98ae6fe0041d50e4d6798dc051